api/src/models/ImageFile.ts

```diff
@@ -71,21 +71,50 @@ class ImageFile {
     }
 
     ocrDerivative() {
-
+        let fs = require("fs");
+        let png = this.derivativePath('ocr/pngs', 'png');
+        let { exec } = require("child_process");
+        if (fs.existsSync(png)) {
+            let tc_cmd = "#{config['textcleaner_path']} #{config['textcleaner_switches']} #{derivative('LARGE')} #{png}"
+            exec(tc_cmd, (error, stdout, stderr) => {
+                if (error) {
+                    console.log(`error: ${error.message}`);
+                    return;
+                }
+                if (stderr) {
+                    console.log(`stderr: ${stderr}`);
+                    return;
+                }
+                console.log(`stdout: ${stdout}`)
+            });
+        }
```

Comment:
At this point, the Ruby code checks to see if the output file was successfully generated and the command line process did not return an error. We should probably do similar validation here.

Reply:
What you added above is a good start, but I think we want to do more than just log a message. We need to fail the build so the problem can be investigated... otherwise jobs may get ingested with no OCR, and we could easily fail to notice. I think the Javascript equivalent of Ruby's `raise` is `throw`.